apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/RetryTest.cs

```diff
@@ -89,41 +90,45 @@ public void ExceptionAfterRetryExhaustedTest()
 
     private static void AssertAttempts(RetryOptions retryOptions, int statusCode, int numOfFailures, int maximumRetries, bool success, params int[] expectedBackOffs)
     {
-        var messageHandler = new ReplayingMessageHandler(VersionHeaderBuilder.HeaderName);
+        var scheduler = new FakeScheduler();
+        var clock = scheduler.Clock;
+        var messageHandler = new ReplayingMessageHandler(VersionHeaderBuilder.HeaderName, clock);
         var service = new FakeStorageService(messageHandler);
         service.HttpClient.MessageHandler.GoogleApiClientHeader = "test/fake";
         service.HttpClient.MessageHandler.NumTries = maximumRetries;
 
         var request = service.Buckets.Get("bucket");
-        var client = new StorageClientImpl(service);
-
-        // Retries for the the failures. Assumed that error code 504 is always included in the predicate for these tests.
-        service.ExpectRequests(request, (HttpStatusCode) 504, numOfFailures - 1);
-        if (numOfFailures > 0)
-        {
-            service.ExpectRequest(request, (HttpStatusCode) statusCode);
-        }
-
-        DateTime startTime = DateTime.UtcNow;
-        if (success)
-        {
-            // Last call is a success.
-            service.ExpectRequest(request, new Bucket());
-
-            client.GetBucket("bucket", new GetBucketOptions { RetryOptions = retryOptions });
-
-            Assert.Equal(expectedBackOffs.Count(), messageHandler.AttemptTimestamps.Count());
-            service.Verify();
-        }
-        else
-        {
-            // The call throws an exception
-            Assert.Throws<GoogleApiException>(() => client.GetBucket("bucket", new GetBucketOptions { RetryOptions = retryOptions }));
-        }
-
-        for (int i = 0; i < messageHandler.AttemptTimestamps.Count; i++)
+        var client = new StorageClientImpl(service, encryptionKey: null, scheduler);
+        scheduler.Run(() =>
```

Comment:
This change is actually very small - it's mostly indentation. All I did was wrap the code in a call to scheduler.Run, and change DateTime.UtcNow to clock.GetCurrentDateTimeUtc()

Reply:
> @anshuldavid13: When this is in, my _hope_ is that in your other code, you can just change "RetryHandler.MarkAsRetriable" to "MarkAsRetriable" everywhere.
Yes Jon I had already created a PR in my fork for the same. Will update it once this is merged.